fitbenchmarking/fitbenchmarking
# Allow access to hessians

#### Description of Work

Fixes #853 

This PR adds the option for users to use hessians for NIST problems with fitting software that accept hessian information. The option `use_hessian` should be set to `True` for hessians to be used. If this option is set with problems/minimizers that are not compatible with using hessians then these problems/minimizers will be skipped. 

To do:
- [x] Get hessians working for nlls and weighted_nlls cost functions
- [x] Add documentation for changes
- [x] Add unit tests for new functions

#### Testing Instructions

1. Run fitbenchmarking with option `use_hessian: True` and check that problems run with reasonable results
2. Run with `use_hessian` set to false and check that results are consistent
3.

Function: Does the change do what it's supposed to?

Tests: Does it pass? Is there adequate coverage for new code?

Style: Is the coding style consistent? Is anything overly confusing?

Documentation: Is there a suitable change to documentation for this change?
